tasks.py

```diff
@@ -196,14 +197,21 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None):
     run("kind load docker-image --name={} metallb/mirror-server:dev-{}".format(name, architecture), echo=True)
 
     run("kubectl delete po -nmetallb-system --all", echo=True)
-    with open("manifests/metallb.yaml") as f:
-        manifest = f.read()
-    manifest = manifest.replace(":main", ":dev-{}".format(architecture))
-    manifest = manifest.replace("imagePullPolicy: Always", "imagePullPolicy: Never")
-    with tempfile.NamedTemporaryFile() as tmp:
-        tmp.write(manifest.encode("utf-8"))
-        tmp.flush()
-        run("kubectl apply -f {}".format(tmp.name), echo=True)
+
+    manifests_dir = os.getcwd() + "/manifests"
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # Copy manifests dir to temp dir.
+        shutil.copytree(manifests_dir, tmpdir, dirs_exist_ok=True)
+
+        with open(tmpdir + "/metallb.yaml") as f:
+            manifest = f.read()
+        manifest = manifest.replace(":main", ":dev-{}".format(architecture))
+        manifest = manifest.replace("imagePullPolicy: Always", "imagePullPolicy: Never")
+        with open(tmpdir + "/metallb.yaml", "w") as f:
+            f.write(manifest)
+            f.flush()
+
+        run("kubectl apply -k {}".format(tmpdir), echo=True)
```

Comment:
Just curious (no strong opinion if using kustomize or not): why use kustomize instead of `kubectl apply -f manifests/namespace.yaml` first and then the metallb.yaml?

I mean, both should do the same now and I'm not sure we want to kustomize something in the dev env via kustomize. What was your reasoning for using kustomize instead of the applying the two yamls (`namespace.yaml` and `metallb.yaml`) present in the manifest folder?

Reply:
I've switched to Kustomize following 514091a2. I'm not sure about the reasoning behind this commit since its message doesn't explain the change, however - yes, we can definitely do `kubectl apply -f` on the relevant manifests only.

My *personal* rationale for this refactor was: assuming the `manifests/` dir keeps changing, it's more generic to operate on the entire dir rather than on individual manifests. Using Kustomize it's safe to do so since only the manifests explicitly mentioned in `kustomizations.yaml` are applied.

Ideally we probably want to deploy MetalLB to the dev environment as similarly as possible to the way users deploy on real clusters. I am not sure what the canonical way for deploying MetalLB is since the [docs](https://metallb.universe.tf/installation/) mention both a Kustomize and a non-Kustomize options. In any case I don't have a strong opinion here besides saying we should have a working dev environment :-)

Reply:
Actually the rationale for 514091a is explained in the PRs description: https://github.com/metallb/metallb/pull/516. Sorry :-|

Reply:
Good point, I agree. And using kustomize may be simpler to maintain (although I doubt the manifest will change much... but who knows), then :ok_hand: